File: src/main/scala/com/pwootage/oc/js/duktape/DuktapeArchitecture.scala
package com.pwootage.oc.js.duktape

import com.pwootage.oc.js._
import li.cil.oc.api.machine.{Architecture, Machine}


/**
  * Spidermonkey
  */
@Architecture.Name("ES5 (duktape)")
class DuktapeArchitecture(_machine: Machine) extends JSArchitectureBase(_machine) {

  override def createEngine(): JSEngine = {
    new DuktapeEngine(this)
  }

  override def setupSandbox(): Unit = {
    //TODO
  }
}

File: src/main/java/com/pwootage/oc/js/duktape/DuktapeArchitecture.kt
package com.pwootage.oc.js.duktape

import com.pwootage.oc.js.JSArchitectureBase
import com.pwootage.oc.js.JSEngine
import li.cil.oc.api.machine.Architecture
import li.cil.oc.api.machine.Machine


/**
 * Spidermonkey
 */
@Architecture.Name("ES5 (duktape)")
class DuktapeArchitecture(_machine: Machine) : JSArchitectureBase(_machine) {

  override fun createEngine(): JSEngine {
    return DuktapeEngine(this)
  }
}
